azkaban-web-server/src/main/java/azkaban/webapp/servlet/LoginAbstractAzkabanServlet.java

```diff
@@ -409,6 +410,10 @@ protected void handleAjaxLoginAction(final HttpServletRequest req,
       cookie.setPath("/");
       resp.addCookie(cookie);
       getApplication().getSessionCache().addSession(session);
+      final Set<Session> sessionsOfSameIP =
+          getApplication().getSessionCache().findSessionsByIP(session.getIp());
+      // Check potential DDoS attack by bad hosts.
+      logger.info(sessionsOfSameIP.size() + " session(s) found from this IP " + session.getIp());
```

Comment:
Can we add the log message when there are more than one sessions per ip?
BTW, not all sessions are registered in SessionCache. Here `LoginAbstractAzkabanServlet.doPost()` sessions are temporarily created but are not registered. Do you know the reason behind it?

Reply:
enrich the logging to have more info.

Temp session is used when user specify username/password in their non-login request. But I do think this is an anti-pattern in terms of security. Let's see if we can deprecate this behavior.